src/net.h

```diff
@@ -361,6 +361,11 @@ class Transport {
 
     /** Return the memory usage of this transport attributable to buffered data to send. */
     virtual size_t GetSendMemoryUsage() const noexcept = 0;
+
+    // 3. Miscellaneous functions.
+
+    /** Whether upon disconnections, a reconnect with V1 is warranted. */
+    virtual bool ShouldReconnectV1() const noexcept = 0;
```

Comment:
6a2dbed1209b18d923da64b019f6f378d3c30695 could make this more generic `ShouldReconnect()` (see comment above).